test/e2e/certmanager/metrics_test.go

```diff
@@ -0,0 +1,211 @@
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, metricsReaderClusterRoleBinding, true)
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, curlPod, true, util.LongTimeout)
+		util.ExpectAllPodsInNamespaceDeleted(ctx, k8sClient, ns)
+	})
+
+	ginkgo.When("workload is admitted", func() {
+		var (
+			clusterQueue *v1beta1.ClusterQueue
+			localQueue   *v1beta1.LocalQueue
+			workload     *v1beta1.Workload
+		)
+
+		ginkgo.BeforeEach(func() {
+			clusterQueue = utiltesting.MakeClusterQueue("").
+				GeneratedName("test-cq-").
+				ResourceGroup(
+					*utiltesting.MakeFlavorQuotas(resourceFlavor.Name).
+						Resource(corev1.ResourceCPU, "1").
+						Resource(corev1.ResourceMemory, "1Gi").
+						Obj(),
+				).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			localQueue = utiltesting.MakeLocalQueue("", ns.Name).
+				GeneratedName("test-lq-").
+				ClusterQueue(clusterQueue.Name).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+
+			workload = utiltesting.MakeWorkload("test-workload", ns.Name).
+				Queue(localQueue.Name).
+				PodSets(
+					*utiltesting.MakePodSet("ps1", 1).Obj(),
+				).
+				RequestAndLimit(corev1.ResourceCPU, "1").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, workload)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, workload, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, localQueue, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		})
+
+		ginkgo.It("should expose quota reserved workload metric", func() {
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, workload)
+
+			ginkgo.By("checking that the quota reserved workload metric is present", func() {
+				expectedMetric := []string{
+					"kueue_quota_reserved_workloads_total",
+					clusterQueue.Name,
+				}
+				expectMetricsToBeAvailable(curlPod.Name, curlContainerName, [][]string{expectedMetric})
+			})
+		})
+	})
+})
+
+func getKueueMetricsSecure(curlPodName, curlContainerName string) ([]byte, error) {
+	metricsOutput, _, err := util.KExecute(ctx, cfg, restClient, config.DefaultNamespace, curlPodName, curlContainerName,
+		[]string{
+			"/bin/sh", "-c",
+			fmt.Sprintf(
+				"curl -s --cacert %s/ca.crt -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" https://%s.%s.svc.cluster.local:8443/metrics",
+				certMountPath,
+				metricsServiceName,
+				config.DefaultNamespace,
+			),
+		})
+
+	return metricsOutput, err
+}
+
+func expectMetricsToBeAvailable(curlPodName, curlContainerName string, metrics [][]string) {
+	gomega.Eventually(func(g gomega.Gomega) {
+		metricsOutput, err := getKueueMetricsSecure(curlPodName, curlContainerName)
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+
+		g.Expect(string(metricsOutput)).Should(utiltesting.ContainMetrics(metrics))
+	}, 10*time.Minute).Should(gomega.Succeed())
```

Comment:
Why 10min? Also, don't use custom timeouts, maybe VeryLongTimeout const (5min) is enough, but I would expect even 45s should be enough?

Reply:
I will keep it to `util.Timeout`